src/expr/src/scalar/mod.rs

```diff
@@ -725,6 +725,7 @@ pub enum EvalError {
     UnterminatedLikeEscapeSequence,
     Parse(ParseError),
     Internal(String),
+    InfinityOutOfDomain(String),
```

Comment:
I'm not sure if this is a granularity that's OK for errors, and whether keeping a string around (in this case, just to report the name of the function) is acceptable. For the inverse functions, a new type of error would have to be created, as `asin` and `acos` have `[-1, 1]` as their domain.

Reply:
Oh yeah, this is totally great. It's definitely fine to allocate in these messages for now; the goal with the enum is just to make it _possible_ to avoid the allocation should it ever be an issue.